### PR TITLE
dns/dnscrypt-proxy: Support specifying relays for anonymous DNS

### PR DESCRIPTION
**Pull request by schreibubi**
Support specifying relays for anonymous DNS

### CODE REVIEW

**Comment by bethebeer** [comment]
This looks great and I think it's important functionality to add. If there's anything I can do to help get this merged, let me know.

**Comment by schreibubi** [comment]
@mimugmail would you have the time to review this? Thanks!

**Comment by mimugmail** [comment]
@schreibubi sorry, took to long for such a quick addition. I tested it and it looks fine.
Would you mind adding a link for available relays in the helptext like here https://github.com/opnsense/plugins/blob/master/dns/dnscrypt-proxy/src/opnsense/mvc/app/controllers/OPNsense/Dnscryptproxy/forms/general.xml#L166

File: dns/dnscrypt-proxy/src/opnsense/mvc/app/controllers/OPNsense/Dnscryptproxy/forms/general.xml [hunk 1]
```diff
@@ -179,4 +179,12 @@
         <allownew>true</allownew>
         <help><![CDATA[Exclude servers from automatic selection. Add any specific server names here if you do not want to use them for any reason.]]></help>
     </field>
+    <field>
+        <id>general.relaylist</id>
+        <label>Relay List</label>
+        <type>select_multiple</type>
+        <style>tokenize</style>
+        <allownew>true</allownew>
+        <help><![CDATA[Set a list of relays. Will be used for relaying to all configured servers.]]></help>
```
Comment (schreibubi) [inline dns/dnscrypt-proxy/src/opnsense/mvc/app/controllers/OPNsense/Dnscryptproxy/forms/general.xml:188]: @mimugmail The help information which you are requesting is already available, or am I missing something?

Comment (schreibubi) [reply dns/dnscrypt-proxy/src/opnsense/mvc/app/controllers/OPNsense/Dnscryptproxy/forms/general.xml:188]: Ah sorry, did not read your comment fully, will add the link to the list of available relays.

Comment (schreibubi) [reply dns/dnscrypt-proxy/src/opnsense/mvc/app/controllers/OPNsense/Dnscryptproxy/forms/general.xml]: Okay, made the change you've requested. Be aware, the I pushed a fixup commit, so probably a squash merge will be required. If I should squash locally and repush instead, then please let me know.

**Review by mimugmail — APPROVED** [review]
Thx, looks fine :)

File: dns/dnscrypt-proxy/pkg-descr [hunk 2]
```diff
@@ -4,6 +4,9 @@ such as DNSCrypt v2 and DNS-over-HTTPS.
 
 Plugin Changelog
 ================
+1.11
```
Comment (fichtner) [inline dns/dnscrypt-proxy/pkg-descr:7]:
```suggestion

1.11
```

**Comment by fichtner** [comment]
Merged, thanks!

File: dns/dnscrypt-proxy/src/opnsense/mvc/app/controllers/OPNsense/Dnscryptproxy/forms/general.xml (hunk 1 above)
Comment (rgnldo) [reply dns/dnscrypt-proxy/src/opnsense/mvc/app/controllers/OPNsense/Dnscryptproxy/forms/general.xml]: I didn't understand how to add the server relays for anonymous DNS.
Add the url link or the server?

Comment (mimugmail) [reply dns/dnscrypt-proxy/src/opnsense/mvc/app/controllers/OPNsense/Dnscryptproxy/forms/general.xml]: No, you insert the name of the relay which you can get from the URL

Comment (rgnldo) [reply dns/dnscrypt-proxy/src/opnsense/mvc/app/controllers/OPNsense/Dnscryptproxy/forms/general.xml]: > 

Ok! Uncheck option Servers DNS-over-HTTPS?
Insert one or multiple relay?

Comment (mimugmail) [reply dns/dnscrypt-proxy/src/opnsense/mvc/app/controllers/OPNsense/Dnscryptproxy/forms/general.xml]: Seems @schreibubi needs to write a doc PR too :)

**Pull request merged**